Instagram/ComposeViewController.m

```diff
@@ -18,44 +20,55 @@ - (void)viewDidLoad {
     self.caption.layer.borderWidth = 2.0f;
     self.caption.layer.borderColor = [[UIColor lightGrayColor] CGColor];
     self.caption.layer.cornerRadius = 5;
+}
 
+- (IBAction)onTap:(id)sender {
     UIImagePickerController *imagePickerVC = [UIImagePickerController new];
     imagePickerVC.delegate = self;
     imagePickerVC.allowsEditing = YES;
-
+    
     if ([UIImagePickerController isSourceTypeAvailable:UIImagePickerControllerSourceTypeCamera]) {
-        imagePickerVC.sourceType = UIImagePickerControllerSourceTypeCamera;
+        UIAlertController *alert = [UIAlertController alertControllerWithTitle:@"Photo" message:@"Would you like to take or select a photo?" preferredStyle:(UIAlertControllerStyleAlert)];
```

Comment:
Nice application of concepts learned to a different usage. I don't think I like the user experience but I love that you thought of this and implemented it.